### PR TITLE
Add SMT regression tests

### PR DESCRIPTION
**Pull request by pkesseli**
Add SMT back-end regression tests for entities with a conversion to bitvectors.

These tests test the bugfix of the SMT2 back-end introduced in https://github.com/diffblue/cbmc/pull/3119 and will only pass once it is merged.

### CODE REVIEW

**Review by martin-cs — COMMENTED** [review]
I'm very much on-board with the idea but I think Daniel had concerns about adding Z3 to the CI set-up.

File: buildspec-windows.yml [hunk 1]
```diff
@@ -64,9 +69,10 @@ phases:
         cd ../..
 
       - |
-        $env:Path = "C:\tools\cygwin\bin;$env:Path"
+        $env:Path = "C:\tools\cygwin\bin;C:\tools\cygwin\bin;c:\tools\z3-4.7.1-x64-win\bin;$env:Path"
```
Comment (tautschnig) [inline buildspec-windows.yml:72]: Duplicate `C:\tools\cygwin\bin` directory

File: regression/cbmc-smt/assert1/test.desc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CORE
+main.c
+--smt2
+activate-multi-line-match
```
Comment (tautschnig) [inline regression/cbmc-smt/assert1/test.desc:4]: Seems unnecessary as no multi-line patterns are being used?

File: regression/cbmc-smt/float1/test.desc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+CORE
+main.c
+--smt2 --fpa
+activate-multi-line-match
```
Comment (tautschnig) [inline regression/cbmc-smt/float1/test.desc:4]: Seems unnecessary?

File: regression/cbmc-smt/union1/test.desc [hunk 4]
```diff
@@ -0,0 +1,8 @@
+CORE
+main.c
+--smt2
+activate-multi-line-match
```
Comment (tautschnig) [inline regression/cbmc-smt/union1/test.desc:4]: Seems unnecessary?

File: regression/cbmc-smt/union1/test.desc [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+^EXIT=10$
```
Comment (tautschnig) [inline regression/cbmc-smt/union1/test.desc:5]: Maybe we should also have at least one test that actually verifies successfully?

**Review by tautschnig — COMMENTED** [review]
Some minor comments below; there are two existing tests in the `regression/cbmc` folder marked "smt-backend" that should be moved to the new `cbmc-smt` folder. And then of course CI needs to be brought to the point where the tests pass ;-)

**Comment by pkesseli** [comment]
I think this is at least partially superseded by https://github.com/diffblue/cbmc/pull/3284. I'm not sure whether it's worth adding the `z3` dependency now just to cover a few more edge cases. Opinions, @tautschnig @kroening ?

**Comment by TGWDB** [comment]
Closing as this is effectively superseded and very out of date. Please reopen if you believe this is erroneous and plan to rebase and update.

**Pull request closed without merging**